locodedb/utils.go

```diff
@@ -0,0 +1,104 @@
+package locodedb
+
+import (
+	"bytes"
+	"compress/gzip"
+	_ "embed"
+	"encoding/csv"
+	"io"
+	"strconv"
+	"sync"
+)
+
+var (
+	//go:embed data/countries.csv.gz
+	countriesData []byte
+
+	//go:embed data/locodes.csv.gz
+	locodesData []byte
+
+	locodeDataOnce sync.Once
+)
+
+func initLocodeData() (err error) {
+	locodeDataOnce.Do(func() {
+		unpackCountriesData(countriesData)
+		unpackLocodesData(locodesData)
+	})
+	return
+}
+
+type locodesCSV struct {
+	locationName string
+	subDivCode   string
+	subDivName   string
+	point        *Point
+	continent    *Continent
+}
+
+func unpackCountriesData(data []byte) map[CountryCode]string {
+	gzReader, err := gzip.NewReader(bytes.NewReader(data))
+	if err != nil {
+		panic(err)
+	}
+	defer gzReader.Close()
+
+	reader := csv.NewReader(gzReader)
+
+	m := make(map[CountryCode]string)
+
+	for {
+		record, err := reader.Read()
+		if err == io.EOF {
+			break
+		} else if err != nil {
+			panic(err)
+		}
+		countryCode, _ := CountryCodeFromString(record[0])
```

Comment:
why ignore err?

Reply:
returning `nil` if error appears is not much better. keeping in mind how much i do not like panics, imo even a panic is better than skipping a val or writing an empty/corrupted val